internal/command/clusters/uninstall.go

```diff
@@ -0,0 +1,218 @@
+		Args:  cobra.MatchAll(cobra.ExactArgs(0)),
+		Run: run(func(ctx context.Context, args []string) error {
+			kubeCfg, err := kubernetes.NewConfig(kubeConfigPath)
+			if err != nil {
+				return err
+			}
+
+			// first, we need to check that secrets containing certs are safe from
+			// garbage collection
+			messagesOwnedSecrets := []string{}
+			secretsClient, err := clients.NewGenericClient[*corev1.Secret, *corev1.SecretList](
+				&clients.GenericClientOptions{
+					RestConfig: kubeCfg,
+					APIPath:    "/api/",
+					Group:      corev1.GroupName,
+					Version:    corev1.SchemeGroupVersion.Version,
+					Kind:       "secrets",
+				},
+			)
+
+			var secretsList corev1.SecretList
+			err = secretsClient.List(ctx, &clients.GenericRequestOptions{}, &secretsList)
+
+			for i := range secretsList.Items {
+				secret := &secretsList.Items[i]
+
+				hasCertificateOwnerRef := false
+				for _, ownerRef := range secret.OwnerReferences {
+					if ownerRef.Kind == "Certificate" {
+						hasCertificateOwnerRef = true
+						break
+					}
+				}
+
+				if hasCertificateOwnerRef {
+					messagesOwnedSecrets = append(messagesOwnedSecrets, fmt.Sprintf("%s/%s has certificate owner ref", secret.Namespace, secret.Name))
+				}
+			}
+
+			// next, check if any certificates are about to be renewed
+			messagesUpcomingRenewals := []string{}
+			upcomingCerts := [][]string{}
+			certificateClient, err := clients.NewCertificateClient(kubeCfg)
+			if err != nil {
+				return fmt.Errorf("error creating certificate client: %s", err)
+			}
+
+			var certificates certmanagerv1.CertificateList
+			err = certificateClient.List(
+				ctx,
+				&clients.GenericRequestOptions{},
+				&certificates,
+			)
+			if err != nil {
+				return fmt.Errorf("error listing certificates: %s", err)
+			}
+			warnBuffer, err := time.ParseDuration("1h")
+			if err != nil {
+				return fmt.Errorf("error parsing duration, this is a bug: %s", err)
+			}
+			for _, c := range certificates.Items {
+				if c.Status.RenewalTime != nil {
+					if time.Now().Add(warnBuffer).After(c.Status.RenewalTime.Time) {
+						diff := c.Status.RenewalTime.Sub(time.Now())
+						messagesUpcomingRenewals = append(
+							messagesUpcomingRenewals,
+							fmt.Sprintf(
+								"%s/%s will be renewed soon (%s)",
+								c.Namespace,
+								c.Name,
+								diff.String(),
+							),
+						)
+						upcomingCerts = append(upcomingCerts, []string{c.Namespace, c.Name})
+					}
+				}
+			}
+
+			// check if any certificates are currently being issued
+			messagesIssuingCerts := []string{}
+			for _, cert := range certificates.Items {
+				issuing := false
+				ready := false
+				for _, cond := range cert.Status.Conditions {
+					if cond.Type == certmanagerv1.CertificateConditionIssuing && cond.Status == "True" {
+						issuing = true
+					}
+					if cond.Type == certmanagerv1.CertificateConditionReady && cond.Status == "True" {
+						ready = true
+					}
+				}
+				if issuing && !ready {
```

Comment:
We don't need the readiness check here, a cert will be ready and issuing when it's being renewed unless it has already expired.